### PR TITLE
`AirbyteLogMessage.stack_trace` for logging messages with related (non-fatal) errors

### PR DESCRIPTION
**Pull request by evantahler**
This PR is inspired by the conversation in https://github.com/airbytehq/airbyte/pull/15668#issuecomment-1234773047. Please read this thread first. The goal of this PR is to allow connectors to pass non-fatal stack traces along with an `AirbyteLogMessage` to be logged by the worker.  This is helpful when the connector catches an exception and handles it, e.g. `log("caught HTTP exception, trying again in 10 seconds", exception);`.  The message and exception will be logged together for debugging.

For example: 

```js
{
  TYPE: "LOG", 
  log: {
    level: "INFO", 
    message: "The connector had a problem, but we are retrying in 10 seconds...",
    stack_trace: "OH NO on line 1\r\n\tBECAUSE OF foo on line 2"
}}
```

Will be rendered in the logs like:
```
2022-09-08 20:38:54 source > The connector had a problem, but we are retrying in 10 seconds...
Stack Trace: OH NO on line 1
	BECAUSE OF foo on line 2
```

This PR replaces https://github.com/airbytehq/airbyte/pull/16317.

## Rationale

* We are ok with adding this one additional property to `AirbyteLogMessage` because this one additional named property's use-case is very clear and unlikely to be misinterpreted.
* We shouldn’t add wildcard objects to the protocol (e.g. `AirbyeteLogMessage.data`) because people will do silly with things that we can’t guarantee - e.g. data.foo could be used for triggering other operations which we can’t promise will work in the future.
* Loggers in general seem to understand `stack_trace` as a first class object. This elevates this concept above a wildcard .data object
* Google says "stack trace" is 2 words, and that’s what we already use for `AirbyteErrorTraceMessage` so we will use the underscore
* This allows `AirbyteNotificaitonTraceMessage`  to remain reserved for notifying UI consumers of connector behavior.

## Notes
As this is a change to the [Airbyte Protocol](https://github.com/airbytehq/airbyte/blob/master/airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml), do not merge this PR until the [Protocol Change Process](https://docs.google.com/document/d/1544cfQST-WX9HQvJUq2borVSOLjRTWAJJJd2nmrJ9Io/edit#) has been completed.

To test this PR, `source-faker` has also been modified to emit additional LOG messages.

We use `stack_trace` (underscore) in the protocol.

### CODE REVIEW

**Comment by evantahler** [comment]
Here's what it looks like in the UI:

**Review by sherifnada — COMMENTED** [review]
My understanding from our IRL convo @evantahler was we wanted to a way to do structured logging, and that adding a generic `data` field would allow us to do that. Why has this goal changed? It seemed like a worthwhile target. It seems the main reason was to prevent consumers from making assumptions about the structure of the logs, but I guess I don't see why that is any more true than it is today? (people can make assumptions about the content of the `message` field) 

I thought the previous direction made sense, so I'm just trying to better understand the change

**Comment by evantahler** [comment]
@sherifnada yeah, this is a reversal.  We ended up deciding that the right path forward was to be a strictly typed as possible with the protocol in order to discourage un-expected use cases.  A single "stack_trace" property we can guarantee, and because it's a known term, expect people will use it only for this purpose.  The "data" blob appeared to be to enticing for others to put structured data into it what would be used downstream (e.g. parse the logs for `{next_step:go}` and use that to power dagster or something similar).

File: airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java [hunk 1]
```diff
@@ -92,12 +92,15 @@ public Stream<AirbyteMessage> create(final BufferedReader bufferedReader) {
   }
 
   private void internalLog(final AirbyteLogMessage logMessage) {
+    final String combinedMessage =
+        logMessage.getMessage() + (logMessage.getStackTrace() != null ? ("\r\nStack Trace: " + logMessage.getStackTrace()) : "");
```
Comment (edgao) [inline airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java:96]: nitpick: `System.lineSeparator()` isntead of hardcoded `\r\n`

**Review by edgao — APPROVED** [review]
nice :shipit: 

no comments about the protocol change itself, looks like it aligns with the slack discussions over the last few days. Minor comment about the implementation but I wouldn't block on it.

**Review by sherifnada — APPROVED** [review]
The rationale seems fair enough

Is there a protocol change log that needs to be updated or something?

**Comment by evantahler** [comment]
> Is there a protocol change log that needs to be updated or something?

There isn't!  I'll add one in a separate PR. There might be some conversation about the version numbers.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Pull request merged**